DatadogRUM/Sources/Instrumentation/AppHangs/AppHangsWatchdogThread.swift

```diff
@@ -35,16 +56,27 @@ internal final class AppHangsWatchdogThread: Thread {
     private let dateProvider: DateProvider
     /// Backtrace reporter for hang's stack trace generation.
     private let backtraceReporter: BacktraceReporting
+    /// The hang's duration threshold to consider it a false-positive.
+    private let falsePositiveThreshold: TimeInterval
     /// An identifier of the main thread required for backtrace generation.
     /// Because backtrace is generated from the watchdog thread, we must identify the main thread to be promoted in `BacktraceReport`. This value is
     /// obtained at runtime, only once and it is cached for later use.
     @ReadWriteLock
     private var mainThreadID: ThreadID? = nil
     /// Telemetry interface.
     private let telemetry: Telemetry
-    /// Closure to be notified when App Hang ends. It will be executed on the watchdog thread.
+    /// Closure to be notified when App Hang starts.
+    /// It is executed on the watchdog thread.
+    @ReadWriteLock
+    internal var onHangStarted: ((AppHang) -> Void)?
+    /// Closure to be notified when App Hang gets cancelled due to possible false-positive.
+    /// It is executed on the watchdog thread.
+    @ReadWriteLock
+    internal var onHangCancelled: ((AppHang) -> Void)?
+    /// Closure to be notified when App Hang ends. It passes the hang and its duration.
+    /// It is executed on the watchdog thread.
     @ReadWriteLock
-    internal var onHangEnded: ((AppHang) -> Void)?
+    internal var onHangEnded: ((AppHang, TimeInterval) -> Void)?
     /// A block called after this thread finished its pass and will become idle.
     @ReadWriteLock
     internal var onBeforeSleep: (() -> Void)?
```

Comment:
/suggestion That makes 5 locks :/ The property wrapper might not fit very well here, unless we use a companion object with a single lock.

Another approach would be to create another read-write lock object that is not a property wrapper and that we can use to lock multiple accesses.

Reply:
There is only one write to each `onHang*` lock during SDK startup on the main thread. Later, SDK only reads them from the watchdog (background) thread.

I could combine these callbacks to `Observer` struct, e.g.:
```swift
thread.hangObserver = .init(
   onStarted: { _ in },
   onCancelled: { _ in }
   onEnded: { _, _ in }
)
```
Alternatively, we can leverage `delegate` model (`thread.delegate = self`). WDYT @maxep ?

Reply:
The good old delegate pattern :) It is quite appropriate in this case 👍 Do you mind trying out?

Reply:
Sure, I changed to delegate 👍. It is `weak var` [which implies "atomic"](https://github.com/apple/swift/pull/1454), so no locks required.